slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/validator/GenesisValidatorRootValidator.java

```diff
@@ -57,16 +70,15 @@ public boolean genesisValidatorRootExists() {
                 handle -> metadataDao.findGenesisValidatorsRoot(handle).isPresent()));
   }
 
-  private boolean validateGvr(final Handle handle, final Bytes32 genesisValidatorsRoot) {
+  private Void findAndInsertGVR(final Handle handle, final Bytes32 genesisValidatorsRoot) {
     final Optional<Bytes32> dbGvr = metadataDao.findGenesisValidatorsRoot(handle);
-    final boolean isValidGvr = dbGvr.map(gvr -> gvr.equals(genesisValidatorsRoot)).orElse(true);
-    if (!isValidGvr) {
-      LOG.warn(
-          "Supplied genesis validators root {} does not match value in database",
-          genesisValidatorsRoot);
-    } else if (dbGvr.isEmpty()) {
+    if (dbGvr.isPresent()) {
+      cachedGenesisValidatorRoot = dbGvr.get();
+    } else {
+      // insert and update local cache
       metadataDao.insertGenesisValidatorsRoot(handle, genesisValidatorsRoot);
+      cachedGenesisValidatorRoot = genesisValidatorsRoot;
     }
-    return isValidGvr;
+    return null;
```

Comment:
nit: This feels to me like mixing the cache update with the database stuff. Would it be better to return the gvr and update the cache in the level above?

Reply:
agreed, refactored the code and renamed the method to convey the intention more clearly as well.